### PR TITLE
[rustdoc] Add `is_object_safe` information for traits in JSON output

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
As asked by @obi1kenobi [here](https://github.com/rust-lang/rust/pull/113241#issuecomment-1868213677).

cc @aDotInTheVoid 
r? @notriddle

### CODE REVIEW

**Review by aDotInTheVoid — CHANGES_REQUESTED** [review]
r? @aDotInTheVoid 

r=me with assertions fixed

File: tests/rustdoc-json/traits/is_object_safe.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#![no_std]
+
+// @has "$.index[*][?(@.name=='FooUnsafe')]"
+// @has "$.index[*][?(@.name=='FooUnsafe')].inner.trait.is_object_safe" false
```
Comment (aDotInTheVoid) [inline tests/rustdoc-json/traits/is_object_safe.rs:4]:
```suggestion
// @is "$.index[*][?(@.name=='FooUnsafe')].inner.trait.is_object_safe" false
```

That way you get a better error message from jsondocck, (and also means this test will fail if this ever somehow mathces to `[true, false]`).

The top assert is redundant, but it's fine to keep or remove at your discression,

**Comment by GuillaumeGomez** [comment]
@notriddle Noted!

@bors r=aDotInTheVoid rollup

**Comment by aDotInTheVoid** [comment]
I’m not on review rotation, but am happy to review rustdoc-json stuff.

(is their a way to tell high-five to only assign me these PR’s, or should i rely on the ping, and then r ? myself?)

**Comment by GuillaumeGomez** [comment]
There is a way, but since it's part of rustdoc, I'm not sure it's a good idea to have different handlings for subparts. In any case you're pinged normally so even though it's more of a hassle, you can self-assign when pinged.

**Pull request merged**